azurerm/resource_arm_automation_module.go

```diff
@@ -0,0 +1,186 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationModule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationModuleCreateUpdate,
+		Read:   resourceArmAutomationModuleRead,
+		Update: resourceArmAutomationModuleCreateUpdate,
+		Delete: resourceArmAutomationModuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"automation_account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"module_link": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"uri": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"hash": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"algorithm": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmAutomationModuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationModuleClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Module creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	accName := d.Get("automation_account_name").(string)
+	contentLink := expandModuleLink(d)
+
+	parameters := automation.ModuleCreateOrUpdateParameters{
+		ModuleCreateOrUpdateProperties: &automation.ModuleCreateOrUpdateProperties{
+			ContentLink: &contentLink,
+		},
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resGroup, accName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, accName, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Automation Module '%s' (resource group %s) ID", name, resGroup)
```

Comment:
>%s [](start = 52, length = 2)

We use `%q` instead of `%s` for name and resource group name.

Reply:
I copied this from the other automation resources which where already part of the provider. Changed the resources in scope of PR but fyi, the others do still have %s